### ISSUE TITLE
Consider adding `import/no-unused-modules`.

### ISSUE THREAD

**Issue by manovotny** [manovotny/eslint-config-get-off-my-lawn#148]
It'd be super handy to use [import/no-unused-modules](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-unused-modules.md), but it just reports too many false positives right now.

Desired config: 

```js
'import/no-unused-modules': [
    'error',
    {
        missingExports: true,
        unusedExports: true,
    }
],
```

However:

* `unusedExports` doesn't take MDX files into account when looking for unused items.
* `missingExports` flags all kinds of config files, cli scripts, etc.
